### ISSUE TITLE
Crash on ill-formed input.

### ISSUE THREAD

**Issue by mocabe** [goessner/mdmath#50]
### Summary

Following code crashes mdmath extension.
```md
>$$

$$
```

### Actual behavior
After allocating huge amount of memory,  extension crashes with error "FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory".

### Steps to reproduce
Save following code as .md and open it with mdmath extension enabled.
```md
>$$

$$
```

### Related links

* [Markdown+Math](https://github.com/goessner/mdmath)
* [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath)
### Environment

```text
  Operating system: Win64
  VSCode version: Insider 1.25.0
  Markdown+Math version: v2.2.1
```

**Comment by goessner** [comment]
Yes ... I can reproduce this ... even with vscode version 1.24.

I need to have a deeper look into this and will come back to you here..

thanks

**Comment by goessner** [comment]
Implemented a bug fix with  [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath).

But now fatal bug with ill behavior is gone with VSCode 1.26.0.

Thanks

**Comment by senioria** [comment]
It seems that this issue reappeared on `VSCode 1.30.2` with `Markdown+Math 2.3.7` ...

**Comment by goessner** [comment]
This is quite a pathological bug only occuring, if 
1. a display formula is opened inside of a blockquote section
2. followed by an empty line outside of the blockquote section
3. closing the display formula on a line followed outside of the blockquote section

i.e
```md
>$$

$$
```

This bug in [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath) is resolved with version 0.5.4. and also resolved therefore with mdmath version 2.3.8 available in a couple of days.
